src/main/webapp/app/exercises/shared/exercise/exercise.service.ts

```diff
@@ -56,12 +56,21 @@ export class ExerciseService {
     validateDate(exercise: Exercise) {
         exercise.dueDateError = exercise.releaseDate && exercise.dueDate ? !exercise.dueDate.isAfter(exercise.releaseDate) : false;
 
-        exercise.assessmentDueDateError =
-            exercise.assessmentDueDate && exercise.releaseDate
-                ? !exercise.assessmentDueDate.isAfter(exercise.releaseDate)
-                : exercise.assessmentDueDate && exercise.dueDate
-                ? !exercise.assessmentDueDate.isAfter(exercise.dueDate)
-                : false;
+        if (exercise.releaseDate && exercise.assessmentDueDate) {
+            if (exercise.dueDate) {
+                exercise.assessmentDueDateError = !exercise.assessmentDueDate.isAfter(exercise.dueDate) && !exercise.assessmentDueDate.isAfter(exercise.releaseDate);
+            } else {
+                exercise.assessmentDueDateError = !exercise.assessmentDueDate.isAfter(exercise.releaseDate);
+            }
+        }
+
+        if (exercise.dueDate && exercise.assessmentDueDate) {
+            exercise.assessmentDueDateError = !exercise.assessmentDueDate.isAfter(exercise.dueDate);
+        }
+
+        if (!exercise.dueDate && exercise.assessmentDueDate) {
```

Comment:
Consider pulling the check for `assessmentDueDate` out? This simplifies all if checks.

Reply:
i refactored the whole part to avoid unnecessary checks while including your feedback